### ISSUE TITLE
Invariant Violation: Element type is invalid: expected a string( for built-in components) or a class/function (for composite components) but got: object. You likely forgot to export your component from the file it's defined in.

### ISSUE THREAD

**Issue by AAGSICON** [facebook/react-native#16332]
### Is this a bug report?

yes
![screenshot_2017-10-12-14-03-09-58](https://user-images.githubusercontent.com/31310379/31490178-9cb0a3c6-af32-11e7-9a5b-cfb0ee44d8ee.png)
[App.js.txt](https://github.com/facebook/react-native/files/1378869/App.js.txt)
[User_List.php.txt](https://github.com/facebook/react-native/files/1378872/User_List.php.txt)

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

(Write your answer here.)

### Steps to Reproduce

(Write your steps here:)

1.
2.
3.

### Expected Behavior

(Write what you thought would happen.)

### Actual Behavior

(Write what happened. Add screenshots!)

### Reproducible Demo

(Paste the link to an example project and exact instructions to reproduce the issue.)

**Comment by bobmulder** [comment]
While this report is minimalistic (so I do understand nobody responded on this one), I do have the same error. I'm not on my workspace so I'll try to explain what happens.

I was using version 0.43.x when I upgraded to 0.49.x. This was the moment my app failed because of dependencies (PropTypes, et cetera), so I tried to reinstall packages with NPM, clear cache, et cetera. The code does NOT recognize components by React Native. PHPStorm does not link the imported component (like it should do), so that seems to be related to this error when running `run-android`. 

At this moment I am planning to start a new project (newest version of RN) and copy my `src` folder into it. But this should not be the right solution huh ;)

**Comment by simoarpe** [comment]
Any updates on this?
I'm facing the same error.
I've updated from `0.48.1` to `0.48.4`.

**Comment by simoarpe** [comment]
I answer my own question.
The fix is fairly simple. 
Open `App.js` file and modify this line
`class YourApp extends Component {` 
into 
`export default class YourApp extends Component<{}> {`

**Comment by bobmulder** [comment]
@simoarpe My editor gives me a warning when using `<{}>`. This seems to be invalid. Is that right?

**Comment by simoarpe** [comment]
Supposedly it should be accurate, that is the same syntax used in the `App.js` when a new project is generated using `react-native init YourApp`.

**Comment by bobmulder** [comment]
Okay I'll check it out when I'm home. Thanks for noticing!

**Comment by dexter21r** [comment]
Adding  `export default `  solved my problem

my code

```
//Import library to help create component
import React from 'react';
import { AppRegistry } from 'react-native';
import Header from './src/components/header';

//Create a Component
const App = () => (
  <Header />
);

//Export App - This line solved my issue
export default App;

//Render it to the device
AppRegistry.registerComponent('albums', () => App);
//albums is project name that we use while creating RN App
```

**Comment by jayeshjain24** [comment]
I am facing the same issue on Linux .
react-native-cli: 2.0.1
react-native: 0.50.3
node -v : v9.2.0
npm -v : 5.5.1

**Comment by HannanShaik** [comment]
I am facing this issue as well. I have tried solutions suggested by @dexter21r and @simoarpe but no luck. I am getting this error not in my root `App.js` but in one of my screens where i am using a `ScrollView`

**Comment by satejsarker** [comment]
@rakeshgithub answer is work for me

**Comment by Johncy1997** [comment]
import React, { Component } from 'react';
import { AppRegistry, FlatList,
StyleSheet, Text, View, Image, Alert, Platform,
TouchableHighLight, 
RefreshControl, TextInput} from 'react-native';
import {addNewTask} from '../actions';

export default class AddComponent extends Component{
    constructor(props)
    {
        super(props);
        this.state = ({
            newTaskName: ''
        });
    }
    render(){
      return(
            <View style={{
                backgroundColor: 'tomato',
                flexDirection: 'row',
                justifyContent: 'flex-end',
                alignItems: 'center',
                height: 64
            }}>
            <TextInput style={{
                height :40,
                width: 200,
                margin: 10,
                padding: 10,
                borderColor: 'white',
                borderWidth: 1,
                color: 'white'
            }}
              keyboardType= 'default'
              placeholderTextColor= 'white'
              placeholder= 'Enter task name'
              autoCapitalize= 'none'
              onChangeText={
                  (text) => {
                      this.setState({
                          newTaskName: text
                      });
                  }
              }
            />
            <TouchableHighLight
                style={{
                    marginRight: 30
                }}
            underlayColor='blue'
            onPress={(event)=>{
                if(!this.state.newTaskName.trim()){
                    return;
                }
                this.props.onClickAdd(this.state.newTaskName);
                //call click event => use "Container"
            }}    
            >
            <Image
                        style={{ width: 35, height: 35 }}
                        source={require('../images/download.png')}
                    />
            </TouchableHighLight>
            </View>
        );
    }
}



![screenshot_1518759528](https://user-images.githubusercontent.com/25324938/36294884-d29da98a-1309-11e8-81ba-2b6dfd97ce02.png)

**Comment by smithaitufe** [comment]
Please what is the solution to this error? I have tried all the suggestions and no one worked for me. 

This is making react native really really unpredictable

**Comment by Guy-Leumi** [comment]
Hi smithaitufe, 
You are having the issue because you are probably exporting addNewTask as default and you importing it as named export.
So change import {addNewTask} from '../actions'; to  import addNewTask  from '../actions';

**Comment by olegwn** [comment]
for me it was a difference between: 

import App from './app'

and

import App from './app/index.js'

(the latter fixed the issue)

**Comment by prabin12** [comment]
i solve this problem by creating index.js file and exporting the file from it and at app.js import file from index.js .
suppose you have to import loginform.js  from ./src/component to app.js then first make index.js file on component folder then export loginform  to index.js  on app.js import loginform from index.js

**Comment by terada46** [comment]
having the same problem today and I solved it just as prabin12 did.   I created index.js in my components folder and export my app from it. 

import App from './App.js';
export { App };

**Comment by Kotdnz** [comment]
For me next solution resolved the issue:
export default class App extends React.Component {

**Comment by Kotdnz** [comment]
The string version working as expected:

export default class App extends React.Component {

**Comment by thiago-molive** [comment]
Tive esse problema quando tentei identar o return do render com o parentese na linha debaixo ex:
render() {
return
(
alguma coisa
)
ele só aceitou com o: 
return (
alguma coisa
)

**Comment by darula-hpp** [comment]
I fixed this error after finding that i have been importing Image from native-base isntead of importing it from react-native.

**Comment by kmtilwani** [comment]
Help Me with this --- https://stackoverflow.com/questions/49892807/react-native-camera-video-capturing-shows-error-on-preview-after-it-gets-capture

**Comment by 129emma** [comment]
I got same issue here. And tried various approaches, still not working. The most akward thing is I import FormInputUI.js to another file use exactaly same way as following and it works in that file. 

react-native: 0.55.0,
node: v8.11.1,
npm: 5.6.0

FormInputUI.js

```
import React from 'react';
import { StyleSheet } from 'react-native';
import { Input, Button } from 'react-native-elements';
import Icon from 'react-native-vector-icons/SimpleLineIcons';

const FormInput = props => {
    const { icon, refInput, ...otherProps } = props
    return (
      <Input
        {...otherProps}
        ref={refInput}
        containerStyle={styles.inputContainer}
        icon={<Icon name={icon} color="#7384B4" size={18} />}
        inputStyle={styles.inputStyle}
        autoFocus={false}
        autoCapitalize="none"
        keyboardAppearance="dark"
        errorStyle={styles.errorInputStyle}
        autoCorrect={false}
        blurOnSubmit={false}
        placeholderTextColor="#7384B4"
      />
    )
}
export default FormInput;
```
Main.js
```
import FormInput from './UI/FormInputUI.js';

class LoginForm extends Component {
  render() {
    const {
      isLoading,
      selectedType,
      emailValid,
      passwordValid,
    } = this.state

    return (
      <ScrollView
			scrollEnabled={false}
          keyboardShouldPersistTaps="handled"
          contentContainerStyle={styles.container}
        >
          <KeyboardAvoidingView
            behavior="position"
            contentContainerStyle={styles.formContainer}
          >
            <View>
              <FormInput
                refInput={input => (this.emailInput = input)}
                value={this.props.email}
                onChangeText={this.onEmailChange.bind(this)}
                placeholder="Email"
                keyboardType="email-address"
                returnKeyType="next"
                displayError={!emailValid}
                errorMessage="Please enter a valid email address"
                onSubmitEditing={() => {
                  this.validateEmail()
                  this.passwordInput.focus()
                }}
              />
              <FormInput
                refInput={input => (this.passwordInput = input)}
                value={this.props.password}
                onChangeText={this.onPasswordChange.bind(this)}
                placeholder="Password"
                secureTextEntry
                returnKeyType="next"
                displayError={!passwordValid}
                errorMessage="Please enter at least 6 characters"
                onSubmitEditing={() => {
                  this.validatePassword()
                }}
              />
              { this.renderError() }
            </View>
          </KeyboardAvoidingView>
        </ScrollView>
    )
  }
}

```


![img_6171](https://user-images.githubusercontent.com/32583512/39662067-8ceedfe6-50b0-11e8-9b88-0fcefa4bbbe7.PNG)

**Comment by 129emma** [comment]
@Johncy1997 Hi, I got the same here, did you sovle it?

**Comment by Sunnysh123** [comment]
Am facing same issue can anyone please help me how to resolve this error
![screen shot 2018-05-10 at 5 03 05 pm](https://user-images.githubusercontent.com/27824755/39896566-34d71f9a-5474-11e8-8c8f-15c3fe27b0be.png)

**Comment by 129emma** [comment]
@Sunnysh123 Hi did you check your package.json to update some libraries? cause my issue is the version conflict between different libraries and React Native.

**Comment by pazcur** [comment]
@simoarpe Thank you! That fixed it but the code I used is slightly different that what you used
` export default class App extends React.Component {`

**Comment by jamils82** [comment]
I solved the issue by importing 'Image' from 'react-native' instead of 'native-base'

**Comment by phsantiago** [comment]
You may importing this way:
```jsx
import { foo } from 'bar'
```
when you should:
```jsx
import foo from 'bar'
```


Double check your imports and exports

**Comment by 0paIescent** [comment]
I'm also currently having this issue, but the weird part is it happens even when all I call is this:
`AppRegistry.registerComponent("AppName", () => <View />);`

It's not an export/import issue since all it is is a View.
Any ideas?

**Comment by 0paIescent** [comment]
Figured it out, apparently since I had my folder that I keep all of my components and such in called app, when I import `App from './app'` expecting it to auto pickup the `index.js` in the root of the folder, it didn't for some reason. But if I rename the folder to src, it works perfectly. My fix was just adding `index.js` to the end of my import.

**Comment by bavly** [comment]
I have the same problem tried everything, 

Does anyone know how to solve it

import React, { Component } from 'react';
import { View } from 'react-native';
//import firebase from 'firebase';
import { Header } from './components/common';
import { LoginForm } from './components/LoginForm';

class App extends Component {
  componentWillMount() {

  }
  render() {
    return (
      <View>
        <Header headerText="Auth" />
        <LoginForm />
      </View>
    );
  }
}

export default App;

**Comment by 0paIescent** [comment]
Did you leave out your render function for privacy reasons or is it empty? I believe it will error out if the return is empty, mine actually won't even build if it's empty. Just trying to cover all the bases.

**Comment by ajaykumarramani** [comment]
Problem with native-base component Drawer

React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined
in, or you might have mixed up default and named imports.
Check your code at App.js:21.
07-19 18:13:38.381  5975  6003 W ReactNativeJS:     in App (at renderApplication.js:33)
07-19 18:13:38.381  5975  6003 W ReactNativeJS:     in RCTView (at View.js:60)
07-19 18:13:38.381  5975  6003 W ReactNativeJS:     in View (at AppContainer.js:102)
07-19 18:13:38.381  5975  6003 W ReactNativeJS:     in RCTView (at View.js:60)
07-19 18:13:38.381  5975  6003 W ReactNativeJS:     in View (at AppContainer.js:122)
07-19 18:13:38.381  5975  6003 W ReactNativeJS:     in AppContainer (at renderApplication.js:32)

import React, { Component } from 'react';
import {
  Text,
} from 'react-native';
import { Drawer, Icon, Button, Container, Header, Content, Left, Right, Body } from 'native-base';
import SideBar from './HomeScreen';


 export default class App extends Component {

  closeDrawer(){
    this._drawer._root.close()
  };
  openDrawer(){
    this._drawer._root.open()
  }; 

  render() {
    return (
      <Drawer
        ref = { (ref) => { this._drawer = ref; } }
        content = { <SideBar navigator={ this._navigator } /> }
        onClose = { () => this.closeDrawer() } >
        <Container>
          <Header>
            <Left>
              <Button transparent onPress={() => this.closeDrawer()}>
                <Icon name='menu' />
              </Button>
            </Left>
            <Body>
              <Text>Home Screen</Text>
            </Body>
            <Right>
              <Icon name='home' />
            </Right>
          </Header>
          <Content>
            <Text> Home Screen </Text>
          </Content>
        </Container>
     </Drawer>
    );
  }
}

**Comment by r3wt** [comment]
@ajaykumarramani i have the same problem after updating to latest version of nativebase.

**Comment by sofianegargouri** [comment]
Same error with:

`/index.js`:

```
import { AppRegistry } from 'react-native'
import App from './app'

AppRegistry.registerComponent('MyApp', () => App);
```

`/app/index.js`:

```
import React from 'react'
import { Text, View } from 'react-native'

export default () => (
  <View>
    <Text>Test</Text>
  </View>
)
```

And here is the error:
```
Invariant Violation: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

This error is located at:
    in RCTView (at View.js:60)
    in View (at AppContainer.js:102)
    in RCTView (at View.js:60)
    in View (at AppContainer.js:122)
    in AppContainer (at renderApplication.js:32)
throwOnInvalidElementType
    ReactNativeRenderer-dev.js:4706:4
createFiberFromElement
    ReactNativeRenderer-dev.js:4663:42
reconcileSingleElement
    ReactNativeRenderer-dev.js:8338:8
reconcileChildFibers
    ReactNativeRenderer-dev.js:8422:14
reconcileChildrenAtExpirationTime
    ReactNativeRenderer-dev.js:8622:8
reconcileChildren
    ReactNativeRenderer-dev.js:8603:6
updateHostComponent
    ReactNativeRenderer-dev.js:9001:22
beginWork
    ReactNativeRenderer-dev.js:9589:10
performUnitOfWork
    ReactNativeRenderer-dev.js:12924:25
workLoop
    ReactNativeRenderer-dev.js:12953:43
renderRoot
    ReactNativeRenderer-dev.js:12996:17
performWorkOnRoot
    ReactNativeRenderer-dev.js:13632:34
performWork
    ReactNativeRenderer-dev.js:13545:26
performSyncWork
    ReactNativeRenderer-dev.js:13506:16
requestWork
    ReactNativeRenderer-dev.js:13392:6
scheduleWorkImpl
    ReactNativeRenderer-dev.js:13259:24
scheduleWork
    ReactNativeRenderer-dev.js:13207:28
scheduleRootUpdate
    ReactNativeRenderer-dev.js:13930:17
_updateContainerAtExpirationTime
    ReactNativeRenderer-dev.js:13966:6
updateContainer
    ReactNativeRenderer-dev.js:13991:8
render
    ReactNativeRenderer-dev.js:14726:35
renderApplication
    renderApplication.js:49:21
run
    AppRegistry.js:102:10
runApplication
    AppRegistry.js:194:26
__callFunction
    MessageQueue.js:351:47
<unknown>
    MessageQueue.js:116:26
__guardSafe
    MessageQueue.js:314:6
callFunctionReturnFlushedQueue
    MessageQueue.js:115:17
```

**Comment by danielland85** [comment]
@ajaykumarramani i have the same issue when use native-base component Drawer. 

render() {

    return (

        <Drawer
        ref={(ref) => { this.drawer = ref; }}
        content={<SideBar {...this.props}/>}
        >
            <View style={styles.container}>
                 <Header left='menu' openDrawer={()=>{this.drawer._root.open();}} title={'abc'}/>
                 <Content>
                      <View style={{flex: 3, backgroundColor:'blue'}} >
                       </View>         
                  </Content>
            </View>
        </Drawer>
    );
  }

"react-native": "0.56.0",
"native-base": "^2.7.1",

**Comment by danielland85** [comment]
i found the solution by downgrade native-base@2.7.0

**Comment by th3knigh7** [comment]
huh, it was a typo after all. if any of you is getting undefined and check render method of 'xxxxxx'  then check the xxxxx module for a typo in the import statement.

**Comment by Lalee10** [comment]
If you are importing a class that you have Exported as default. e.g
**'export default ClassName';**

Then you will get this error.
So instead of importing like this **import { ClassName } from './ClassName'**

Use

**import ClassName from './ClassName'**

**Comment by thomasphilibert** [comment]
Hi all,

I have the same problem. I tried all of your solutions but downgrading React from 16.4.2 to 16.2.0 only solve my issue.

**Comment by AlexanderSutul** [comment]
Just make sure that your imports are in needed order, I had the same problem and after fix with imports all started to work good.

**Comment by magicismight** [comment]
这

**Comment by andreyrk** [comment]
In my case it happened when I was importing List from react-native-elements to use it with FlatList.

However, it seems react-native-elements no longer has List as a component since 1.0.0BETA4. Changing List to View fixed it for me.

**Comment by CatherineLiyuankun** [comment]
I solve it by change 
```
export default App;
```
to 
```
module.exports = App;
```

**Comment by iqbinder** [comment]
import React, {Component} from 'react';
import {Text, View, TouchableOpacity, Image, TextInput} from 'react-native';
import styles from '../styles/search-web-screen-styles.js';
import LinearGradient from 'react-native-linear-gradient';
import {Surface} from 'gl-react-native';
import { Blur } from "gl-react-blur";

const GL = require("gl-react");

const shaders = GL.Shaders.create(
  {
          textOverImage: {
            frag:`
            precision highp float;
varying vec2 uv;

uniform sampler2D img;
uniform sampler2D imgBlurred;
uniform sampler2D txt;

const vec2 shadowCenter = vec2(0.5, 0.9);
const vec2 shadowSize = vec2(0.6, 0.2);
float shadow () {
return 0.8 * smoothstep(1.0, 0.2, distance(uv / shadowSize, shadowCenter / shadowSize));
}
float monochrome (vec3 c) {
return 0.2125 * c.r + 0.7154 * c.g + 0.0721 * c.b;
}
vec3 textColor (vec3 bg) {
return vec3(step(monochrome(bg), 0.6));
}

void main () {
vec4 bg = mix(texture2D(img, uv), texture2D(imgBlurred, uv), shadow());
vec4 fg = vec4(textColor(texture2D(imgBlurred, shadowCenter).rgb), 1.0);
float fgFactor = 1.0 - texture2D(txt, uv).r;
gl_FragColor = mix(bg, fg, fgFactor);
}`
          }
  }
);

class HelloBlue extends React.Component
{
  render()
  {
    const { img , width, height } = this.props;
    return <GL.Node shader={shaders.textOverImage}>
  <GL.Uniform name="img">
    {img}
  </GL.Uniform>
  <GL.Uniform name="imgBlurred">
    <Blur factor={20} passes={6} width={width} height={height}>
      {img}
    </Blur>
  </GL.Uniform>
  <GL.Uniform name="txt">
    <Text style={styles.TextInputStyleClass}>Hello world</Text>
  </GL.Uniform>
</GL.Node>;
  }
}


export default class WebSearchScreen extends Component
{
  render()
  {
    return(
      <View style={styles.container}>
          <View style={styles.view_top_bg}>
              <TouchableOpacity style={styles.cross_btn_bg}
                onPress = {() => this.props.navigation.goBack()}>
                  <Image source={require('../assets/icons/cross_1x.png')} />
              </TouchableOpacity>
          </View>

          <View style={styles.view_bottom_bg}>
                <LinearGradient start={{x: 1.0, y: 0.5}} end={{x: 0.5, y: 1.0}}
                                locations={[0.1,0.4,1.0]}
                                colors={['#66c6ff', '#68e3ff', '#ed89ff']}
                                style={styles.LinearGradientStyle} >
                    <TextInput
                        placeholder="Tap to type..."
                        placeholderTextColor="#dbdbdb"
                        underlineColorAndroid='transparent'
                        style={styles.TextInputStyleClass}/>
                </LinearGradient>
          </View>

          <Surface style={styles.view_center_bg}>
              <HelloBlue img={"https://i.imgur.com/wxqlQkh.jpg"} />
          </Surface>


      </View>
    )
  }
}


I am getting the same error in this class . please help me to solve this , I am not getting any link or solution of it.

**Comment by kibernetika** [comment]
It's import problem. When I used:
in cart.js
`const CartItem = require("./cart-item");`
in cart-item.js
`export default CartItem `
I did have this problem, but when I change import:
in cart.js
`import CartItem from "../cart/cart-item";`
all works fine!

**Comment by vreality64** [comment]
**TL;DR**
You should use `import Module from '...'` & `export default Module` pair.
And `const Module = require(...)` is same with `import * as Module from '...'`.

---

It's not matter of react native. It's matter of javascript importing mechanism (esm, commonjs)
Simple solution is to check how to importing React library. Please follow one of below two instruction types.

I think it will help.

### 1. Synthetic import
```ts
import * as React from 'react'

// -> should import module with `* as`
import * as Module from 'third-party-module'

// In this case, it's also works.
// but it's not good to use `import` and `require` together
const Module = require('third-party-module')
```

### 2. Not synthetic import
```ts
import React from 'react'

// -> should import default
import Module from 'third-party-module'
```

**Comment by EdemOne** [comment]
> If you are importing a class that you have Exported as default. e.g
> **'export default ClassName';**
> 
> Then you will get this error.
> So instead of importing like this **import { ClassName } from './ClassName'**
> 
> Use
> 
> **import ClassName from './ClassName'**

This solved it for me

**Comment by krisrandall** [comment]
A slight variation that I'm including for any others that also hit it.

For me, the this code was causing that same error.
**This fails:**
```
    import { View, Text } from 'react-native-elements';

    return (
      <View>
        <Text style={{color: 'red'}}>
          and red
        </Text>
      </View>
    );
```

(although bizarrely, if I changed the View to a Text it worked)

The fix for me was to **import from `react-native` not from `react-native-elements`**.

**This works :**
```
  import { View, Text } from 'react-native';

    return (
      <View>
        <Text style={{color: 'red'}}>
          and red
        </Text>
      </View>
    );
```

**Comment by EphraimMwanda** [comment]
i have been having the same error, i tried all the above solutions and even went to just making a very simple app and discovered that the error only popped up when i was setting up my redux, when i added        //// `<Provider store={store}>`
`      <TodoApp />`
`<Provider>`

the problem might be due to the latest redux package but then again i m not to experienced in react-native to give a solid conclusion.

please assist me further on my issue

**Comment by EphraimMwanda** [comment]
my mistake!!! i was importing my Provider from 'redux' instead from 'react-redux' once i fixed that issue the error stopped showing up
`import { Provider } from 'redux' // WRONG!!`
i corrected the imports by:
`import { Provider } from 'react-redux' //CORRECT!!`

after i fixed that the error stopped showing up!

happy coding people!

**Comment by KunwarVSuryavanshi** [comment]
Removing the curly braces from the name of the function or the class, which I am importing solved the error for me.
Like-
import NewHeader from './components/Header.js';
instead of
import {NewHeader} from './components/Header.js';

**Comment by alexeykomov** [comment]
My reason of getting this error was that I've forgotten to add arguments section for function expression. 
It was
```
const Comp = (
    <View style={{ justifyContent: 'space-between', flexDirection: 'row' }}/>
);
```
Instead of 
```
const Comp = () => (
    <View style={{ justifyContent: 'space-between', flexDirection: 'row' }}/>
);
```

Then `Comp` was rendered as child of other component.
Hope this helps.

**Comment by paulopacitti** [comment]
> I am facing the same issue on Linux .
> react-native-cli: 2.0.1
> react-native: 0.50.3
> node -v : v9.2.0
> npm -v : 5.5.1

I'm having the same issue, and it's the same react-native version. Renaming with "export default" or importing correctly is not the problem. Any news? If I update it, it should work?

**Comment by jeffreyongcay** [comment]
https://stackoverflow.com/questions/34130539/uncaught-error-invariant-violation-element-type-is-invalid-expected-a-string

**Comment by SORNVENGE** [comment]
in file index.js the first i wrote like that:
AppRegistry.registerComponent(appName, () =><App/>);
then i got the same error like above so i have changed it into
=>AppRegistry.registerComponent(appName, () =>App);

**Comment by TimmyGOGO** [comment]
The same issue. I'm trying to pass redux store from first index.js to the next Components. The problem is when I'm passing the store to a simple Component, it's all right, but when I try to pass the store to decorated component, which is being exported like this:
`export default connect(
   mapToStateProps, 
   mapDispatchToProps
)(Component);`
then I get this error. I can't understand why is this happening. Is this because of "complex" store prop that shouldn't be passed to decorated Components?

**Comment by nh83012001** [comment]
This isn't the most helpful error message. my issue was in the import 

import Provider from 'react-redux';

instead of 

import { Provider } from 'react-redux';

**Comment by Lalee10** [comment]
When you "export default" something then in your imports you can name it anything you want and you don't need to use curly braces around the import for example:

export default class A

Then when importing it, you have to follow this syntax

import A from '#filepath';

But if you are exporting the class A like this:

export class A

then you have to import it using the curly braces and the exact name

import { A } from '#filepath';

If you have a lot of exports from a single file e.g:

export class A
export class B

Then you can import them like:

import * as myClasses from '#filepath';
To access class A: myClasses.A
To access class B: myClasses.B

I was getting this error when I was not putting curly braces around my imports when they were not being exported as default

**Comment by fadilnatakusumah** [comment]
damn, @igorchru was right. It's all about attention. you see, i got same problem, turn out I was importing Card component from 'react-native' instead 'react-native-elements'. And my problem is solved.

pay attention to your code guys, stay sharp

**Comment by phoenixxd** [comment]
some kind of whitespace that was not the standard white space was left by code I copied from the internet. I removed everything between component tags and it started working.
